src/cpp/session/modules/SessionHelp.R

```diff
@@ -799,3 +799,15 @@ options(help_type = "html")
 {
    .rs.followHelpTopic(url)
 })
+
+.rs.addFunction("Rd2HTML", function(file, package = "", Rdmacros = "")
+{
+   tf <- tempfile(); on.exit(unlink(tf))
+   macros <- suppressWarnings(tools:::initialRdMacros(Rdmacros))
+   tools::Rd2HTML(file, out = tf, package = package, macros = macros)
+   lines <- readLines(tf, warn = FALSE)
+   lines <- sub("R Documentation</td></tr></table>", "(preview) R Documentation</td></tr></table>", lines)
+   if (nzchar(package))
+      lines <- sub('img src="../help/figures/', sprintf('img src="/library/%s/html/figures/', package), lines, fixed = TRUE)
```

Comment:
Maybe instead of that we could serve e.g. `/preview?file={dirname(file)}/figures/lifecycle-archived.svg`.